demos/speech_recognition_demo/python/utils/deep_speech_seq_pipeline.py

```diff
@@ -0,0 +1,50 @@
+#
+# Copyright (C) 2019-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+# This file is based in part on deepspeech_openvino_0.5.py by Feng Yen-Chang at
+# https://github.com/openvinotoolkit/open_model_zoo/pull/419, commit 529805d011d9b405f142b2b40f4d202bd403a4f1 on Sep 19, 2019.
+#
+from copy import deepcopy
+
+from utils.audio_features import AudioFeaturesSeqPipelineStage
+from utils.rnn_seq_pipeline import RnnSeqPipelineStage
+from utils.ctc_decoder_seq_pipeline import CtcDecoderSeqPipelineStage
+
+
+class DeepSpeechSeqPipeline:
+    def __init__(self, model, lm=None, beam_width=500, max_candidates=None,
+            profile=None, ie=None, device='CPU', online_decoding=False):
+        """
+            Args:
+        model (str), filename of IE IR .xml file of the network
+        lm (str), filename of LM (language model)
+        beam_width (int), the number of prefix candidates to retain during decoding in beam search (default 500)
+        max_candidates (int), limit the number of returned candidates; None = do not limit (default None)
+        profile (dict): a dict with pre/post-processing parameters, see profiles.py
+        ie (IECore or None), IECore object to run NN inference with.  Default is to use ie_core_singleton module.
+            (default None)
+        device (str), inference device for IE, passed here to 1. set default device, and 2. check supported node types
+            in the model load; None = do not check (default 'CPU')
+        online_decoding (bool), set to True to return partial decoded text after every input data piece (default False)
+        """
+        assert profile is not None, "profile argument must be provided"
+        self.p = deepcopy(profile)
+        self.mfcc_stage = AudioFeaturesSeqPipelineStage(profile)
+        self.rnn_stage = RnnSeqPipelineStage(model, profile=profile, ie=ie, device=device)
+        self.ctc_stage = CtcDecoderSeqPipelineStage(lm=lm, profile=profile, beam_width=beam_width,
+                max_candidates=max_candidates, online=online_decoding)
+
+    def recognize_audio(self, audio, sampling_rate, finish=True):
+        """
+        Run a segment of audio through ASR pipeline.
+        Use finish=True (default) to run recognition once for a whole utterance.
+        Use finish=False for online recognition, provinging data cut into segments and getting updated recognition
+        result after each segment. Set finish=True for the last segment to get the final result and reset pipeline state.
+        """
+        if audio is not None:
+            if abs(sampling_rate - self.p['model_sampling_rate']) > self.p['model_sampling_rate'] * 0.1  or  (audio.shape + (1,))[1] != 1:
+                raise ValueError("Input audio file should be {} kHz mono".format(self.p['model_sampling_rate']/1e3))
```

Comment:
`sampling_rate` is passed here just to let `recognize_audio()` verify it. This is strange approach. Move the check to the reset
```py
        assert sample_width == 2, "Only 16-bit WAV PCM supported"
        assert compression_type == 'NONE', "Only linear PCM WAV files supported"
        assert channel_num == 1, "Only mono WAV PCM supported"
```

Reply:
`sampling_rate` checking is done inside because supplying an audio with wrong sampling rate may more-or-less work, but with impaired quality. With other parameters wrong, it either works, or it doesn't.
If some user just copies our files and invokes ASR from his code with a wrong sampling rate (say, the usual 44.1 kHz or 48 kHz instead of 16 kHz), he'll just have somewhat impaired quality, and wouldn't realize he's doing it wrong.

Reply:
There is nothing coupling audio and sampling_rate passed to `recognize_audio()`. `recognize_audio()` has no way to verify the two objects are connected. Basically `recognize_audio()` requires an audio and just a number around 16 kHz. One still may pass wrong audio and after noticing that there is that check put unrelated constant. At least that's how I usually do stuff when something doesn't work :)

Reply:
I think it is obvious from the arg name that it's the sampling rate, and not just some unrelated number. If someone looks at the code that checks its value, he'd notice the error message "Input audio file should be {} kHz mono" suggesting that it's the sampling rate of the audio. And if he doesn't look inside, he's expected to provide values to `audio` and `sampling_rate` according to their names and their usage in the main file.

Reply:
Ok. Pleas update speech_recognition_offline_demo here as well to follow the same approach.

Reply:
OK.